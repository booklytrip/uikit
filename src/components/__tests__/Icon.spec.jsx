import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('Icon', () => {
    it('renders without crashing', () => {
        mount(<Icon />);
    });
    
    it('should call onClick event if clicked', () => {
        const spy = jest.fn();
        const icon = mount(<Icon onClick={ spy } />)
        icon.simulate('click');
        
        expect(spy).toHaveBeenCalled();
    });
    
    describe('Snapshots', () => {
        it('renders loading spinner', () => {
            const rendered = renderer.create(<Icon name="spinner" loading />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders disabled icon', () => {
            const rendered = renderer.create(<Icon name="settings" disabled />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with custom size', () => {
            const rendered = renderer.create(<Icon name="settings" size="mini" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders formated as a link', () => {
            const rendered = renderer.create(<Icon name="settings" link />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders formatted to appear circular', () => {
            const rendered = renderer.create(<Icon name="settings" circular />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders formatted to appear bordered', () => {
            const rendered = renderer.create(<Icon name="settings" bordered />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
        
        it('renders with different color', () => {
            const rendered = renderer.create(<Icon name="settings" color="red" />);
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });    
});
