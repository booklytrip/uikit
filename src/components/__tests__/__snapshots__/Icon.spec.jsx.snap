exports[`Icon Snapshots renders disabled icon 1`] = `
<i
  className="icon settings disabled"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders formated as a link 1`] = `
<i
  className="icon settings link"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders formatted to appear bordered 1`] = `
<i
  className="icon settings bordered"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders formatted to appear circular 1`] = `
<i
  className="icon settings circular"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders loading spinner 1`] = `
<i
  className="icon spinner loading"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders with custom size 1`] = `
<i
  className="icon mini settings"
  onClick={undefined}
  style={undefined} />
`;

exports[`Icon Snapshots renders with different color 1`] = `
<i
  className="icon red settings"
  onClick={undefined}
  style={undefined} />
`;
